openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1235,6 +1256,60 @@ def update_library_collection_components(
     return collection
 
 
+def set_library_component_collections(
+    library_key: LibraryLocatorV2,
+    component: Component,
+    *,
+    collection_keys: list[str],
+    created_by: int | None = None,
+    # As an optimization, callers may pass in a pre-fetched ContentLibrary instance
+    content_library: ContentLibrary | None = None,
+) -> Component:
+    """
+    It Associates the component with collections for the given collection keys.
+
+    Only collections in queryset are associated with component, all previous component-collections
+    associations are removed.
+
+    If you've already fetched the ContentLibrary, pass it in to avoid refetching.
+
+    Raises:
+    * ContentLibraryCollectionNotFound if any of the given collection_keys don't match Collections in the given library.
+
+    Returns the updated Component.
+    """
+    if not content_library:
+        content_library = ContentLibrary.objects.get_by_key(library_key)  # type: ignore[attr-defined]
+    assert content_library
+    assert content_library.learning_package_id
+    assert content_library.library_key == library_key
+
+    # Note: Component.key matches its PublishableEntity.key
+    collection_qs = authoring_api.get_collections(content_library.learning_package_id).filter(
```

Comment:
I think we can put this inside the `set_collections` API, check my [question](https://github.com/openedx/openedx-learning/pull/238#discussion_r1796273313)

Reply:
Commented [here](https://github.com/openedx/openedx-learning/pull/238#discussion_r1796441135).

Reply:
Thanks